torch/_dynamo/decorators.py

```diff
@@ -81,6 +82,36 @@ def skip(fn=None):
     return fn
 
 
+class set_stance(_DecoratorContextManager):
+    """
+    Decorator, context manager, function to set the current stance of the compiler.
+
+    Stances documented in corresponding function in torch/compiler/__init__.py
+    """
+
+    _dynamo_forbidden = True
+
+    def __init__(self, stance: str) -> None:
+        self.stance = stance
+        self.prev = _set_stance(stance)
```

Comment:
Won't calling this in the constructor mean that, defining a function:
```
@set_stance("foo")
def bar():
   pass
```
Will change the global stance even if `bar()` is unused?

Reply:
`__call__` will run in decorator mode, which resets the stance state to the previous state (this logic was taken from the `torch.set_grad_enabled` implementation).